extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalog.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+    /**
+     * Subscribe to a table from the Iceberg catalog. Initially the latest snapshot will be loaded, but the output table
+     * will be updated as new snapshots are added to the table.
+     *
+     * @param tableIdentifier The table identifier to load
+     * @return The loaded table
+     */
+    @SuppressWarnings("unused")
+    public Table subscribeTable(@NotNull final TableIdentifier tableIdentifier) {
+        return readTableInternal(tableIdentifier, null, true);
+    }
+
+    private Table readTableInternal(
+            @NotNull final TableIdentifier tableIdentifier,
+            @Nullable final String snapshotId,
+            final boolean isRefreshing) {
+        // Validate that the user is not trying to subscribe to a snapshot.
+        Assert.eqFalse(isRefreshing && (snapshotId != null),
+                "Must not specify a snapshot ID when subscribing to a table.");
+
+        // Load the table from the catalog
+        final org.apache.iceberg.Table table = catalog.loadTable(tableIdentifier);
+
+        final Snapshot snapshot;
+        final Schema schema;
+        // Do we want the latest or a specific snapshot?
+        if (snapshotId == null) {
+            snapshot = table.currentSnapshot();
+            schema = table.schema();
+        } else {
+            // Load the specific snapshot and retrieve the schema for that snapshot
+            snapshot = table.snapshot(snapshotId);
+            schema = table.schemas().get(snapshot.schemaId());
+        }
+
+        // Load the partitioning schema
+        final org.apache.iceberg.PartitionSpec partitionSpec = table.spec();
+
+        // Convert the Iceberg schema to a Deephaven TableDefinition
+        final TableDefinition tableDefinition = IcebergTools.fromSchema(schema, partitionSpec);
+
+        // Build a parquet instructions object
+        final ParquetInstructions instructions = ParquetInstructions.builder()
+                .setSpecialInstructions(s3Instructions)
+                .build();
+
+        final String description;
+        final TableLocationKeyFinder<IcebergTableLocationKey> keyFinder;
+        final TableDataRefreshService refreshService;
+        final UpdateSourceRegistrar updateSourceRegistrar;
+
+        if (partitionSpec.isUnpartitioned()) {
+            // Create the flat layout location key finder
+            keyFinder = new IcebergFlatLayout(snapshot, fileIO, instructions);
+        } else {
+            final String[] partitionColumns =
+                    partitionSpec.fields().stream().map(PartitionField::name).toArray(String[]::new);
+
+            // Create the partitioning column location key finder
+            keyFinder = new IcebergKeyValuePartitionedLayout(
+                    snapshot,
+                    fileIO,
+                    partitionColumns,
+                    instructions);
+        }
+
+        if (isRefreshing) {
+            refreshService = TableDataRefreshService.getSharedRefreshService();
```

Comment:
Don't forget, when we want to make this a real thing that actually works, we want to also give the target user a mechanism for pushing snapshots at us.